util-collection/src/main/scala/sbt/internal/util/Settings.scala

```diff
@@ -811,9 +811,13 @@ trait Init[ScopeType]:
    */
   sealed trait Keyed[S, A1] extends Initialize[A1]:
     def scopedKey: ScopedKey[S]
-    def transform: S => A1
-
     override final def dependencies = scopedKey :: Nil
+    private[sbt] override def processAttributes[A2](init: A2)(f: (A2, AttributeMap) => A2): A2 =
+      init
+  end Keyed
+
+  final class GetValue[S, A1](val scopedKey: ScopedKey[S], val transform: S => A1)
```

Comment:
Could you add Scala comment explaining the overall motivation of why this class is needed for later record plz? In general, in the future if someone comes along and try to refactor this, they should know to test the heap usage etc.

Reply:
This class is not new. What I did here is to move the implementation of `apply`, `mapReferenced`,  `mapConstant` etc from the abstract `Keyed` to the more concrete `GetValue` and `KeyedInitialize`. Thus the implementation is more specialized and it does not wrap `KeyedInitialize` into `GetValue` or compose the `identity` function anymore. This is kind of obvious when you look at the implementation and I don't think it needs a comment.

Apart from that I will add comments in other parts of the code where I use caching.